### PR TITLE
Upload Jesse Bellemare's data

### PR DESCRIPTION
**Pull request by rudeboybert**

### CODE REVIEW

**Comment by rudeboybert** [comment]
Hey @djkelly00, you're on the right track! I made some minor commits via 4 commits above; while the commit message should be informative enough I encourage you to click on all 4 messages to see what I changed. 

Please pull these changes before working further, and don't forget to build the package often

One requested change: please rename the datasets that the package will load to something more informative than `plot_data01` and `plot_data02`. You want the names to be informative, but not too long either. You might need to [agonize](https://www.facebook.com/Rmemes0/posts/there-are-only-two-hard-things-in-computer-science-cache-invalidation-and-naming/2020837754635172/) a little tho lol

When you think you're done, please tag me for review (top right of this page under reviewers)

File: R/data.R [hunk 1]
```diff
@@ -120,4 +120,32 @@
...
+#' Tree diameter was measured at 1.4 meters high above the ground.
+#' @docType data
+#' @format 
+#' \describe{
+#'  \item{When}{Data collected from 2010-2015 in the fall semesters.}
+#'  \item{Size}{20 x 50 m and divided into 10 subplot modules of 110 m. Five 10 x 10 modules along central 50 m axis.} 
+#'  \item{Comments}{comments}
+#'  \item{People}{Data obtained from Jesse Bellemare and Smith College students from BIO364-365 courses.}
+#' }
+"plot_data01"
```
Comment (rudeboybert) [inline R/data.R:137]: Consider renaming data to have a more informative name

**Comment by beanumber** [comment]
This is shaping up. Please try to follow this style guide:

https://style.tidyverse.org/syntax.html#object-names

In particular, please keep all variable names in [snake_case](https://en.wikipedia.org/wiki/Snake_case)

File: data-raw/plot_data.R [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+View(newplot_data01)
+
+plot_data01 <- newplot_data01
+
+usethis::use_data(plot_data01, overwrite = TRUE)
+
+
+#Plot Data 2
+library(readxl)
```
Comment (rudeboybert) [inline data-raw/plot_data.R:51]: Basic principle of good coding: [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) Don't repeat yourself.  You already loaded package above

File: data-raw/plot_data.R [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+newplot_data02 <- newplot_data02 %>%
+  relocate("Notes:", .after = "Diameter(cm)")
+
+View(newplot_data02)
+
+plot_data02 <- newplot_data02
+
+usethis::use_data(plot_data02, overwrite = TRUE)
+
+#Run this in console for documentation functionality
```
Comment (rudeboybert) [inline data-raw/plot_data.R:82]: this gets run automatically when you build package, so you can remove

File: R/data.R [hunk 4]
```diff
@@ -120,4 +120,32 @@
...
+"maple_sap"
+
+#' MacLeish Data Plot 1
+#' @description Collected data on change in tree diameter (in centimeters)
+#' for Hemlock Woolly Adelgid dominated areas on the Western side of MacLeish. 
+#' Tree diameter was measured at 1.4 meters high above the ground.
+#' @docType data
+#' @format 
+#' \describe{
+#'  \item{When}{Data collected from 2010-2015 in the fall semesters.}
```
Comment (rudeboybert) [inline R/data.R:132]: There seem to be missing variables that aren't documented

File: R/data.R [hunk 5]
```diff
@@ -120,4 +120,32 @@
...
+#'  \item{Size}{20 x 50 m and divided into 10 subplot modules of 110 m. Five 10 x 10 modules along central 50 m axis.}
```
Comment (rudeboybert) [inline R/data.R:133]: See Ben Baumer's comment about variable naming conventions.

**Review by rudeboybert — COMMENTED** [review]
@djkelly00 Great first attempt! Now to drill down on details. Please address all feedback in this review; you can check off "Resolve Conversation" as you go to keep track. This will likely take a few rounds of code review since this is our first time

File: data-raw/plot_data.R [hunk 6]
```diff
@@ -0,0 +1,88 @@
+library(readxl)
+
+#Uploading data into R
+plot_data01 <- read_excel("data-raw/MacLeish Data_Plot MC-01.xlsx")
+plot_data02 <- read_excel("data-raw/MacLeish Data_Plot MC-02.xlsx")
+
+usethis::use_data(plot_data01, overwrite = TRUE)
```
[the remark this answers is not in the document]
Comment (djkelly00) [reply data-raw/plot_data.R]: Are you suggesting I move "usethis::use_data(plot_data01, overwrite = TRUE)" to the end of the .R script?

**Comment by rudeboybert** [comment]
Hey @djkelly00, I just made four changes above, please pull them. Click the commit hashes (Ex 7643c70, 8112d4d) to see what the relevant codes changes. In the meanwhile, a couple more things you're going to need to fix:

``` r
library(macleish)
tree_diameter1
#> # A tibble: 1,020 x 8
#>    module   tag species      position  semester year  diameter notes            
#>     <dbl> <dbl> <chr>        <chr>     <chr>    <chr> <chr>    <chr>            
#>  1      1  2001 Betula lenta sub-cano… Fall     2015  NA       Alive in 2009, d…
#>  2      1  2001 Betula lenta sub-cano… Fall     2014  NA       Alive in 2009, d…
#>  3      1  2001 Betula lenta sub-cano… Fall     2013  NA       Alive in 2009, d…
#>  4      1  2001 Betula lenta sub-cano… Fall     2012  NA       Alive in 2009, d…
#>  5      1  2001 Betula lenta sub-cano… Fall     2011  NA       Alive in 2009, d…
#>  6      1  2001 Betula lenta sub-cano… Fall     2010  NA       Alive in 2009, d…
#>  7      1  2002 Tsuga canad… sub-cano… Fall     2015  10.5     <NA>             
#>  8      1  2002 Tsuga canad… sub-cano… Fall     2014  10.4     <NA>             
#>  9      1  2002 Tsuga canad… sub-cano… Fall     2013  10.5     <NA>             
#> 10      1  2002 Tsuga canad… sub-cano… Fall     2012  10.4     <NA>             
#> # … with 1,010 more rows
tree_diameter2
#> # A tibble: 850 x 7
#>    module   tag species          semester year    diameter notes
#>     <dbl> <dbl> <chr>            <chr>    <chr>   <chr>    <chr>
#>  1      1  2171 Tsuga canadensis Fall     " 2016" 28.2     <NA> 
#>  2      1  2171 Tsuga canadensis Fall     "2012"  27.4     <NA> 
#>  3      1  2171 Tsuga canadensis Fall     "2011"  27.2     <NA> 
#>  4      1  2171 Tsuga canadensis Fall     "2010"  27.1     <NA> 
#>  5      1  2171 Tsuga canadensis Fall     "2009"  27       <NA> 
#>  6      1  2172 Betula lenta     Fall     " 2016" 24.5     <NA> 
#>  7      1  2172 Betula lenta     Fall     "2012"  23.2     <NA> 
#>  8      1  2172 Betula lenta     Fall     "2011"  22.9     <NA> 
#>  9      1  2172 Betula lenta     Fall     "2010"  22.5     <NA> 
#> 10      1  2172 Betula lenta     Fall     "2009"  22.2     <NA> 
#> # … with 840 more rows
```

<sup>Created on 2021-07-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

Note how `year` and `diameter` are `chr` character variables and not numeric? If they're not numeric, its going to be very hard for the user to make plots with. Could you edit your `plot_data.R` some more to ensure these are numeric variables and NAs are dealt with appropriately?

**Comment by rudeboybert** [comment]
Hey @djkelly00, looks good! Note that after running `usethis::use_data(tree_diameter1, overwrite = TRUE)` and `usethis::use_data(tree_diameter2, overwrite = TRUE)` in `data-raw/plot_data.R` you want to also push the results`data/.rda` files as well. I'm going to merge this into `SURF21` now. If Prof. Bellemare answers us and wants changes, we can do that later. congrats!

**Pull request merged**